dataverse-webapp/src/main/java/edu/harvard/iq/dataverse/search/index/SearchPermissionsFinder.java

```diff
@@ -0,0 +1,207 @@
+package edu.harvard.iq.dataverse.search.index;
+
+import edu.harvard.iq.dataverse.DataverseRoleServiceBean;
+import edu.harvard.iq.dataverse.RoleAssigneeServiceBean;
+import edu.harvard.iq.dataverse.persistence.DvObject;
+import edu.harvard.iq.dataverse.persistence.dataset.Dataset;
+import edu.harvard.iq.dataverse.persistence.dataset.DatasetVersion;
+import edu.harvard.iq.dataverse.persistence.dataset.DatasetVersion.VersionState;
+import edu.harvard.iq.dataverse.persistence.dataverse.Dataverse;
+import edu.harvard.iq.dataverse.persistence.group.Group;
+import edu.harvard.iq.dataverse.persistence.user.AuthenticatedUser;
+import edu.harvard.iq.dataverse.persistence.user.Permission;
+import edu.harvard.iq.dataverse.persistence.user.RoleAssignee;
+import edu.harvard.iq.dataverse.persistence.user.RoleAssignment;
+
+import javax.ejb.Stateless;
+import javax.inject.Inject;
+
+import java.time.Instant;
+import java.util.ArrayList;
+import java.util.HashMap;
+import java.util.HashSet;
+import java.util.List;
+import java.util.Map;
+import java.util.Set;
+import java.util.logging.Logger;
+
+/**
+ * Determine whether items should be searchable.
+ */
+@Stateless
+public class SearchPermissionsFinder {
+
+    private static final Logger logger = Logger.getLogger(SearchPermissionsFinder.class.getCanonicalName());
+
+    private RoleAssigneeServiceBean roleAssigneeService;
+    private DataverseRoleServiceBean rolesSvc;
+    
+    // -------------------- CONSTRUCTORS --------------------
+    
+    @Deprecated
+    public SearchPermissionsFinder() {
+        // JEE requirement
+    }
+    
+    @Inject
+    public SearchPermissionsFinder(RoleAssigneeServiceBean roleAssigneeService, DataverseRoleServiceBean rolesSvc) {
+        this.roleAssigneeService = roleAssigneeService;
+        this.rolesSvc = rolesSvc;
+    }
+
+    // -------------------- LOGIC --------------------
+    
+    /**
+     * @return A list of strings on which Solr will JOIN to enforce permissions
+     * @todo Should we make a PermStrings object? Probably.
```

Comment:
Well todo's doesn't work in this project, If we want to change it in the future I would just add an issue to github with description and just remove TODO, it's better to fill up issue board than code imo.

Reply:
Well. It is a object. That is leftover from old code. I will remove it.